Restore tests for old application-gateway as a new project (#9394)
kyma-project/kyma: tests/application-gateway-legacy-tests/README.md

# Application Gateway Tests

## Overview

This project contains the acceptance tests for the Kyma Application Gateway in the legacy mode.

## Prerequisites

The project requires Go 1.8 or higher.

## Usage

Environment parameters used by the tests:

| Name | Required | Description | Default | Example value |
|------|----------|---------|-------------|-----------------|
| **APPLICATION** | Yes | Name of the Application to test | None | `my-application` |
| **NAMESPACE** | Yes | Namespace in which the test Application will operate | None | `kyma-integration` |
| **MOCK_SERVICE_PORT** | Yes | Number of the port used by the mock service created by the test | None | `8080` |

> | **TEST_EXECUTOR_IMAGE** | No | Image name of the test executor created by the Helm test | Version matching the Helm test image | `user/my-image:1.0.0` |

Reviewer: ```suggestion
| **TEST_EXECUTOR_IMAGE** | No | Name of the test executor image created by the Helm test | Version matching the Helm test image | `user/my-image:1.0.0` |
```

# Application Gateway Tests

## Overview

> This project contains the acceptance tests for the Kyma Application Gateway.

Reviewer: ```suggestion
This project contains the acceptance tests for the Kyma Application Gateway in the legacy mode.
```

## Prerequisites

The project requires Go 1.8 or higher.

## Usage

Environment parameters used by the tests:

> | Name | Required | Default | Description | Possible values |

Reviewer: ```suggestion
| Name | Required | Default | Description | Example value |
```

> | Name | Required | Default | Description | Possible values |

Reviewer: See the [guidelines for tables](https://github.com/kyma-project/community/blob/master/guidelines/content-guidelines/04-formatting.md#tables).
Although we have some inconsistencies in our documentation, our tables generally follow the order: 
```
| Parameter      | Required | Description   | Default value |       {PROFILE} value       | Example value   |
|----------------|:--------:|---------------|---------------|-----------------------------|-----------------|
| {Param's name} | {Yes/No} | {Description} |      None     | {Value for the given prof.} | {Example value} |
```
For consistency's sake, we should stick to the same order when creating tables.

Author: At first I will switch Default and Description columns. I will lookup for PROFILE values then.

Reviewer: You don't need to apply all the columns. Choose the ones you need, just stick to the order. The profile value refers to tables like [this one](https://sap.sharepoint.com/sites/100093/_layouts/15/Doc.aspx?sourcedoc={d9b81bfd-650d-4e42-9f62-43220886f783}&action=edit&wd=target%28SIG%20Content.one%7C0a2f8356-73ae-4f4a-adf6-0a3617f99e51%2FRelease%20notes%201.16%20%28input%5C%29%7C22d7a1e1-38b5-449d-8b8d-b42c3745632c%2F%29).

|------|----------|---------|-------------|-----------------|

> | **APPLICATION** | Yes | None | The name of the Application to test | `my-application` |

Reviewer: ```suggestion
| **APPLICATION** | Yes | None | Name of the Application to test | `my-application` |
```

> | **NAMESPACE** | Yes | None | The Namespace in which the test Application will operate | `kyma-integration` |

Reviewer: ```suggestion
| **NAMESPACE** | Yes | None | Namespace in which the test Application will operate | `kyma-integration` |
```
We tend to skip the definite article at the beginning of the description in such columns for improved readability.

> | **MOCK_SERVICE_PORT** | Yes | None |  Port number used by the mock service created by the test | `8080` |

Reviewer: ```suggestion
| **MOCK_SERVICE_PORT** | Yes | None | Number of the port used by the mock service created by the test | `8080` |
```

> | **TEST_EXECUTOR_IMAGE** | No | Same version as the Helm test image  | Image name of the test executor created by the Helm test  | `user/my-image:1.0.0` |

Reviewer: ```suggestion
| **TEST_EXECUTOR_IMAGE** | No | Version matching the Helm test image  | Image name of the test executor created by the Helm test  | `user/my-image:1.0.0` |
```

> 

Reviewer: ```suggestion
```
